experimental/shared/src/main/scala/zio/prelude/experimental/package.scala

```diff
@@ -0,0 +1,24 @@
+package zio.prelude
+
+package object experimental extends AddMultiplyShapeSyntax with DivideShapeSyntax with SubtractShapeSyntax {
+
+  object classic {
+
+    import zio.prelude.classic._
+
+    type Semiring[A] = Annihilation[A] with DistributiveMultiply[A] {
+      type Addition[x] <: CommutativeMonoid[x]
+      type Multiplication[x] <: Identity[x]
+    }
+    type Ring[A]     = Semiring[A] with SubtractShape[A] {
+      type Addition[x] <: AbelianGroup[x]
+      type Multiplication[x] <: Identity[x]
+    }
+    type Field[A]    = Ring[A] with DivideShape[A] {
```

Comment:
No matter how we change this PR, we should keep in mind our end goal is this level of modularity. ❤️